dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs

```diff
@@ -22,7 +23,11 @@ public class VolatileMemoryStore : IMemoryStore
     /// <inheritdoc/>
     public Task CreateCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
     {
-        if (!this._store.TryAdd(collectionName, new ConcurrentDictionary<string, MemoryRecord>()))
+        try
+        {
+            this._store.TryAdd(collectionName, new ConcurrentDictionary<string, MemoryRecord>());
```

Comment:
The name of the method called here (TryAdd) implies that the call WON'T throw.

Maybe it's in THAT method that the change should be made???